docs/_source/tutorials/notebooks/training-llm-mistral-sft.ipynb

```diff
@@ -0,0 +1,602 @@
+    "from typing import Dict, Any\n",
+    "\n",
+    "def extract_background_prompt_response(text: str) -> Dict[str, Any]:\n",
+    "    '''Extract the anthropic prompt from a prompt and response pair.'''\n",
+    "    start_prompt = text.find(\"<human>:\")\n",
+    "    end_prompt = text.rfind(\"<bot>:\")\n",
+    "    # Background is anything before the first <human>:\n",
+    "    background = text[:start_prompt].strip()\n",
+    "    # Prompt is anything between the first <human>: (inclusive) and the last <bot>: (exclusive)\n",
+    "    prompt = text[start_prompt: end_prompt].strip()\n",
+    "    # Response is everything after the last <bot>: (inclusive)\n",
+    "    response = text[end_prompt:].strip()\n",
+    "    return {\"background\": background, \"prompt\": prompt, \"response\": response}\n",
+    "\n",
+    "\n",
+    "data = data.map(extract_background_prompt_response, input_columns=\"text\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can trivially convert this dataset into `FeedbackRecord` instances, and add them to the `dataset`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "records = [\n",
+    "    rg.FeedbackRecord(\n",
+    "        fields={\n",
+    "            \"background\": sample[\"background\"],\n",
+    "            \"prompt\": sample[\"prompt\"],\n",
+    "            \"response\": sample[\"response\"],\n",
+    "        },\n",
+    "    )\n",
+    "    for sample in data\n",
+    "]\n",
+    "dataset.add_records(records)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have all the records locally, we want to push them to the Argilla server as well. Only then will we be able to see them in the Argilla UI."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.push_to_argilla(\"oig-30k\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![training-llm-mistral-sft](../../_static/tutorials/training-llm-mistral-sft/training-llm-mistral-sft.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## (Optional) Annotate instruction samples\n",
+    "If you are using your own proprietary data, data from an unreliable source, or require on your annotators to provide responses, then you must perform data annotation. However, for the purposes of this tutorial, we will assume that all data is high quality and skip this step."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up the ArgillaTrainer for Supervised Finetuning\n",
+    "Next, we have to set up the [ArgillaTrainer](https://docs.argilla.io/en/latest/practical_guides/fine_tune.html#supervised-finetuning). First off, we will load the Mistral 7B `model` and `tokenizer`.\n"
```

Comment:
maybe add some context about the configs used when loadng the model? Why do you change the pad token etc

Reply:
Covered in 06e5319750a3f4f5947bc6ed6675a9c9d5437afd